### PR TITLE
Add a legend position dropdown to the chart config

### PR DESCRIPTION
**Pull request by jmuharsky**
The chart sidebar UX now includes a combo box for choosing the Legend position.

### CODE REVIEW

File: client/components/widget/data_viz/gviz/chart-wrapper-service.js [hunk 1]
```diff
@@ -51,6 +51,19 @@ explorer.components.widget.data_viz.gviz.ChartWrapperService = function($http,
   this.GvizChartWrapper_ = GvizChartWrapper;
 
   /**
+   * A list of legend positions for GViz charts.
+   * @export {!Array.<!string>}
+   */
+  this.legendPositions = [
```
Comment (cmccoy) [inline client/components/widget/data_viz/gviz/chart-wrapper-service.js]: Maybe make this a constant?

**Comment by cmccoy** [comment]
What's the logic for adding this to Explorer? Legend position can be modified within the chart editor. Is there a system for deciding what chart configuration options are promoted to the widget editor?

**Comment by jmuharsky** [comment]
The general problem is that chart functionality not present in the Chart Dialog (such as "legend" on candlestick charts) is overwritten when the dialog is used.  Going forward, after the new sidebar UX is in place, charts will eventually have rich editors for most/all of the properties.  Also considered merging the two objects, and that's still a possibility for a future fix, but there are concerns that will require additional effort beyond "use npm object_merge" because there are some settings that are removed from the Chart Dialog because they are explicitly configured to be not present, which we'd want to "destroy" in a merge.

This specific PR works around the most aggregious customer-reported issue, which is the JSON-configured legend disappearing from charts that don't support it in the popup dialog.

File: client/components/widget/data_viz/gviz/chart-config-directive.html [hunk 2]
```diff
@@ -33,11 +41,14 @@
         </div>
       </div>
 
-      <div class="sidebar-item">
-        <div class="sidebar-item-label">column span</div>
+      <div class="sidebar-item" ng-hide="ngModel.chart.chartType == 'Table'">
```
Comment (cmccoy) [inline client/components/widget/data_viz/gviz/chart-config-directive.html]: Recommend using `p3rf.perfkit.explorer.models.ChartType.TABLE` here.

**Comment by cmccoy** [comment]
OK, thanks for explaining. Could you add a comment with some of this information or create an issue and reference it in the code?
Otherwise, LGTM with some nits.

**Pull request merged**